### ISSUE TITLE
Certbot plugins does not list dns-porkbun - failed renewal

### ISSUE THREAD

**Issue by Stinj** [infinityofspace/certbot_dns_porkbun#87]
**Describe the bug**
`The requested dns-porkbun plugin does not appear to be installed.`

**To Reproduce**
```
sudo snap install --classic certbot
sudo snap set certbot trust-plugin-with-root=ok
sudo snap install --edge certbot-dns-porkbun
sudo snap connect certbot:plugin certbot-dns-porkbun
```

**Expected behavior**
`sudo certbot plugins` should list `dns-porkbun`, but does not even after removal and reinstall of snaps.

**Certbot command**
```
sudo snap connect certbot:plugin certbot-dns-porkbun
sudo certbot plugins
```

**Versions (please complete the following version information):**
 - certbot: 2.11.0
 - certbot_dns_porkbun: 0.9 (edge)

**Error message**
The requested dns-porkbun plugin does not appear to be installed.

**Additional context**
```
sudo snap connections
Interface           Plug            Slot                         Notes
content[certbot-1]  certbot:plugin  certbot-dns-porkbun:certbot  manual
```
`/var/log/letsencrypt/letsencrypt.log` shows the following lines up top:
```
2024-10-03 20:55:16,207:DEBUG:urllib3.connectionpool:http://localhost:None "GET /v2/connections?snap=certbot&interface=content HTTP/1.1" 200 792
2024-10-03 20:55:16,392:DEBUG:certbot._internal.main:certbot version: 2.11.0
2024-10-03 20:55:16,392:DEBUG:certbot._internal.main:Location of certbot entry point: /snap/certbot/3834/bin/certbot
2024-10-03 20:55:16,392:DEBUG:certbot._internal.main:Arguments: ['--dry-run', '--preconfigured-renewal']
2024-10-03 20:55:16,392:DEBUG:certbot._internal.main:Discovered plugins: PluginsRegistry(PluginEntryPoint#apache,PluginEntryPoint#manual,PluginEntryPoint#nginx,PluginEntryPoint#null,PluginEntryPoint#standalone,PluginEntryPoint#webroot)
```

The PluginEntryPoint#null seems to be the culprit and gets filtered out.

Renewal was working just fine before this.

**Comment by infinityofspace** [comment]
Thank you for your bug report.

Version `0.9` was not yet been released and is only available in snap's edge channel. Therefore, the direct solution to the problem would be to switch to version `0.8`, which is available in the `candidate` and `beta` channels.
The dev aka testing version, which is available in the edge channel as version `0.9`, changes the target python version to `3.12` (core24 instead of core20) to prepare for the certbot `3.0` release also with `core24`. 
However, since certbot has not yet released the version `3.0`, only the old version with `core20` is available as stable version. The two different versions are not compatible with each other and therefore the plugin is not recognized correctly.
But there is already a certbot version `2.12.0.dev0` which was published today in the `edge` channel and has already been built with `core24`. The certbot_dns_porkbun plugin dev/testing version `0.9` is compatible with this certbot version. So you have to install the new certbot version to use the certbot_dns_porkbun plugin dev/testing version `0.9`.

However, please note that the edge channel may contain possible problems and errors and is not intended for stable usage.

**Comment by Stinj** [comment]
Thank you for the elaborate reply. Switching to `candidate` was successful. Closing this issue as behavior was as intended.
